src/commands/rover/BindGroupCommand.js

```diff
@@ -66,45 +73,51 @@ class BindGroupCommand extends Command {
         )
       }
 
-      if (ranksString != null) {
-        const ranks = []
+      // Converts the ranks into an object for quick lookup for existence
+      const existingRanks = group.ranks.reduce((obj, item) => (obj[item] = true, obj), {})
+
+      if (ranksString !== undefined) {
         const unparsedRanks = ranksString.split(',')
         for (const rank of unparsedRanks) {
           const rangeMatch = rank.match(/(\d+)-(\d+)/)
-          const rankNumber = parseInt(rank, 10)
-
+          
           if (rangeMatch) {
-            const start = parseInt(rangeMatch[1], 10)
-            const stop = parseInt(rangeMatch[2], 10)
+            const start = Util.clamp(parseInt(rangeMatch[1], 10), 0, 255)
+            const stop = Util.clamp(parseInt(rangeMatch[2], 10), 0, 255)
 
             if (start && stop) {
               for (let i = start; i <= stop; i++) {
-                ranks.push(i)
+                if (!existingRanks[i]) group.ranks.push(i)
               }
             }
-          } else if (rankNumber != null) {
-            ranks.push(rankNumber)
+          } else if (rank.match(/[\d]+/)) {
+            const rankNumber = Util.clamp(parseInt(rank), 0, 255)
+            if (!existingRanks[rank]) group.ranks.push(rankNumber)
+          } else {
+            return msg.reply(`:no_entry_sign: You have attempted to bind an invalid rank (\`${rank}\`) for group (\`${groupId}\`). Ranks should be a whole number.`)
```

Comment:
Does this break Virtual groups, like Asset, which can accept an unbounded number?

Reply:
On Line 61 ranks is initialized to an empty array if there is no pre-existing bind, similar to the previous functionality of setting groups.ranks to an empty array if no rank number is specified. This PR removes various instances of setting ranks to an empty array, and instead puts it in one place.

If you mean using a command like `!bindrank <role> Asset`, without specifying a number, the resulting binding saved will still include an empty ranks array.

Reply:
No, I mean that in this code the number is calmped to 0..255. The asset virtual group needs to be able to accept a "rank" (in this case, argument) that can be larger than that.

Reply:
This should now be solved in the latest commit.

Reply:
Did you test this and confirm that both regular and vritual groups work with this new behavior?

Reply:
Yes. Running the command as `!bindrank <role> 1:266 Asset:4564646544` had produced the following in the guild data json file.
`"groupRankBindings":[{"role":"537664071999422474","groups":[{"id":"1","ranks":[255]},{"id":"Asset","ranks":[4564646544]}]}]`

Running the command a second time as `!bindrank <role> 2:266 Asset:6756765,2323` resulted in the following:
`"groupRankBindings":[{"role":"537664071999422474","groups":[{"id":"1","ranks":[255]},{"id":"Asset","ranks":[2323,6756765,4564646544]},{"id":"2","ranks":[255]}]}]`